catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -110,7 +113,8 @@
             "Multithreaded"}, multiValued = false, description =
             "Number of threads to use when dumping. Setting "
                     + "this value too high for your system can cause performance degradation.")
-    int multithreaded = 20;
+    int multithreaded = Runtime.getRuntime()
+            .availableProcessors();
```

Comment:
@pklinef (@coyotesqrl) I ran some tests with (with the paging updates) to see the effect that multithreading/batch size has on the dump command. At a minimum I'd say we should adjust batch size to 512. The thread count i'm not so sure about. 

I ingested 38,809 records and dumped them using variations of multithreading and batch size.

------
[Chart Data Here](https://gist.github.com/rzwiefel/f97c8b7fe76117c2a481851aaaf1a904)
(X: batch size, Y: Dump time in seconds)
![screenshot 2017-09-08 13 56 13](https://user-images.githubusercontent.com/9013416/30231077-0d179796-949e-11e7-8c6c-531d9a85671e.png)

Reply:
Both dials - particularly thread count - are very sensitive to environment. However, I think 512 seems like a good default value and your analysis supports that. I'd probably stick with a conservative default thread count proportionate to available cores and keep it in the smaller side. One or two per core. This will leave resources available for other connections.

Reply:
What datasets did you use?  How much memory did the JVM have?  I have seen very different impacts adjusting threads and batch size depending on if the ingested files create large or small metacards.

Reply:
@pklinef I used the wikipedia dataset, so very large metacard's  (which I would presume would have benefited the most from a large amount of threads). Left the JVM at 4GB aka the **default** setting, which means our commands should probably reflect that default.